src/main/java/graphql/normalized/VariableAccumulator.java

```diff
@@ -28,8 +29,12 @@ public VariableAccumulator(@Nullable VariablePredicate variablePredicate) {
         valueWithDefinitions = new ArrayList<>();
     }
 
-    public boolean shouldMakeVariable(ExecutableNormalizedField executableNormalizedField, String argName, NormalizedInputValue normalizedInputValue) {
-        return variablePredicate != null && variablePredicate.shouldMakeVariable(executableNormalizedField, argName, normalizedInputValue);
+    public boolean shouldMakeVariable(ExecutableNormalizedField executableNormalizedField, QueryAppliedDirective queryAppliedDirective, String argName, NormalizedInputValue normalizedInputValue) {
+        if (queryAppliedDirective != null) {
+            return variablePredicate != null && variablePredicate.shouldMakeVariable(executableNormalizedField, queryAppliedDirective, argName, normalizedInputValue);
+        } else {
+            return variablePredicate != null && variablePredicate.shouldMakeVariable(executableNormalizedField, argName, normalizedInputValue);
+        }
```

Comment:
this internal call back is the same - when queryAppliedDirective is null - it must be just for a field

Reply:
done